File: cloud/src/Signal.Core/Contacts/IContact.cs
using System;

namespace Signal.Core.Contacts;

public interface IContact
{
    string EntityId { get; }

    string ChannelName { get; }

    string ContactName { get; }

    string? ValueSerialized { get; }

    DateTime TimeStamp { get; }
    string? Metadata { get; }
}

File: web/apps/app/src/contacts/IContact.ts

export type ContactMetadataV1 = {
    Version: 1;
    ProcessSameValue?: boolean;
    PersistHistory?: boolean;
};

export default interface IContact {
    entityId: string;
    channelName: string;
    contactName: string;
    timeStamp: Date | undefined;
    valueSerialized: string | undefined;
    metadata?: ContactMetadataV1;
}
